src/SimpleWMSBody.hpp

```diff
@@ -0,0 +1,128 @@
+#include "../../../src/cs-scene/CelestialBody.hpp"
+
+#include <VistaKernel/GraphicsManager/VistaOpenGLDraw.h>
+#include <VistaOGLExt/VistaBufferObject.h>
+#include <VistaOGLExt/VistaGLSLShader.h>
+#include <VistaOGLExt/VistaTexture.h>
+#include <VistaOGLExt/VistaVertexArrayObject.h>
+
+#include <glm/gtc/type_ptr.hpp>
+
+namespace csp::simpleWmsBodies {
+
+struct Properties {
+  cs::utils::Property<bool> mEnableInterpolation = true;
+  cs::utils::Property<bool> mEnableTimespan      = false;
+};
+
+/// This is just a sphere with a texture, attached to the given SPICE frame. The texture should be
+/// in equirectangular projection.
+class SimpleWMSBody : public cs::scene::CelestialBody, public IVistaOpenGLDraw {
+ public:
+  SimpleWMSBody(std::shared_ptr<cs::core::GraphicsEngine> const& graphicsEngine,
+      std::shared_ptr<cs::core::SolarSystem> const& solarSystem, std::string const& sCenterName,
+      std::string sTexture, std::string const& sFrameName, double tStartExistence,
+      double tEndExistence, std::vector<Wms> tWms,
+      std::shared_ptr<cs::core::TimeControl> timeControl,
+      std::shared_ptr<Properties>            properties = nullptr);
+  ~SimpleWMSBody() override;
+
+  /// The sun object is used for lighting computation.
+  void setSun(std::shared_ptr<const cs::scene::CelestialObject> const& sun);
+
+  /// Interface implementation of the IntersectableObject, which is a base class of
+  /// CelestialBody.
+  bool getIntersection(
+      glm::dvec3 const& rayOrigin, glm::dvec3 const& rayDir, glm::dvec3& pos) const override;
+
+  /// Interface implementation of CelestialBody.
+  double     getHeight(glm::dvec2 lngLat) const override;
+  glm::dvec3 getRadii() const override;
+
+  /// Interface implementation of IVistaOpenGLDraw.
+  bool Do() override;
+  bool GetBoundingBox(VistaBoundingBox& bb) override;
+
+  std::vector<Wms> getWms();
+  Wms              getActiveWms();
+
+  void setActiveWms(Wms wms);
+  void setActiveWms(std::string wms);
+
+  std::vector<timeInterval> getTimeIntervals();
+
+ private:
+  std::shared_ptr<cs::core::GraphicsEngine> mGraphicsEngine;
+  std::shared_ptr<cs::core::SolarSystem>    mSolarSystem;
+
+  std::vector<Wms> mWms;
+  Wms              mActiveWms;
+  std::mutex       mWmsMutex;
+  bool             wmsInitialized = false;
+
+  std::shared_ptr<VistaTexture> mTexture;
+  std::shared_ptr<VistaTexture> mWmsTexture;
+  std::shared_ptr<VistaTexture> mOtherTexture;
+  std::shared_ptr<VistaTexture> mDefaultTexture;
+  std::string                   mDefaultTextureFile;
+  int                           mDefTextWidth;
+  int                           mDefTextHeight;
+  std::string                   mRequest;
+  std::string                   mFormat;
+  int                           mIntervalDuration;
+  int                           mTextureWidth;
+  int                           mTextureHeight;
+  int                           mPreFetch; /// Amount of textures that gets prefetched in every direction.
```

Comment:
These parameters seem somehow to be a code duplication from the `Wms` (or `WMSConfig`) struct. Can we use this here?

Reply:
Yes, I've changed it to use instead of `mTextureWidth`, `mTextureHeight` and `mPreFetch`.